### PR TITLE
perf: optimize Deno.writeFile

### PR DESCRIPTION
**Pull request by littledivy**
WIP

### CODE REVIEW

File: runtime/js/40_write_file.js [hunk 1]
```diff
@@ -33,15 +45,35 @@
       abortHandler = () => core.tryClose(cancelRid);
       options.signal[abortSignal.add](abortHandler);
     }
-    try {
-      await core.opAsync("op_write_file_async", {
-        path: pathFromURL(path),
-        data,
+    const path = pathFromURL(pathOrURL);
+    let fsFile = openFdCache.get(path);
+    const append = options.append ?? false;
```
Comment (aapoalas) [inline runtime/js/40_write_file.js:50]: `options.append === true`

File: runtime/js/40_write_file.js [hunk 2]
```diff
@@ -33,15 +45,35 @@
...
+    const create = options.create ?? true;
```
Comment (aapoalas) [inline runtime/js/40_write_file.js:51]: `options.create !== false`

File: runtime/js/40_write_file.js (hunk 1 above)
Comment (aapoalas) [reply runtime/js/40_write_file.js:50]: Though, these would change observable behaviour, presumably changing throw errors into silent defaults.

**Review by lucacasonato — COMMENTED** [review]
I don't get this. Writing to the same fd multiple times will just keep appending data... this seems very strange to me.

**Review by lucacasonato — COMMENTED** [review]
Why is this faster than `op_write_file_async`?

**Comment by littledivy** [comment]
yikes... def not how it should be done

**Pull request closed without merging**